components/camel-csv/src/main/java/org/apache/camel/dataformat/csv/CsvMarshaller.java

```diff
@@ -53,15 +53,15 @@ protected CsvMarshaller(CSVFormat format) {
     public static CsvMarshaller create(CSVFormat format, CsvDataFormat dataFormat) {
         org.apache.camel.util.ObjectHelper.notNull(format, "CSV format");
         org.apache.camel.util.ObjectHelper.notNull(dataFormat, "CSV data format");
+        // Get header columns before potential clearing
+        String[] headers = format.getHeader();
```

Comment:
The old code uses dataFormat.getHeader() and you removed this code why?

Reply:
The reason is quite simple, `CsvDataFormat.getActiveFormat` is setting header information into `CsvFormat`. With this change, you'll catch both cases header set via `DataFormat` or via `CsvFormat`.